### PR TITLE
tart: init at 1.6.0

### LINKED ISSUE #228159: Package request: tart
**Project description**

**Metadata**

* homepage URL: https://tart.run/
* source URL: https://github.com/cirruslabs/tart/
* license: [fair source license](https://fair.io/?a)
* platforms: darwin

### PR DESCRIPTION
**Pull request by emilytrau**
###### Description of changes

macOS VMs on Apple Silicon to use in CI and other automations https://tart.run

Resolves #228159

###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- [x] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes (or backporting 23.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Review by 06kellyjac — COMMENTED** [review]
Looks good, some thoughts on the fairsource attrset

**Review by techknowlogick — COMMENTED** [review]
Thanks for reviving this PR. I found later on that I had an external dependency that was installed outside of nix that probably would need to be included too https://github.com/cirruslabs/softnet

**Comment by 06kellyjac** [comment]
As a note [0.38.0](https://github.com/cirruslabs/tart/releases/tag/0.38.0) from 2023-02-20 is their last [AGPL v3.0 release](https://github.com/cirruslabs/tart/blob/0.38.0/LICENSE)

Probably no benefit packaging that version too but I wonder if anyone would fork it in the future

File: pkgs/applications/virtualization/tart/default.nix [hunk 1]
```diff
@@ -0,0 +1,36 @@
+{ lib
+, stdenvNoCC
+, fetchzip
+}:
+stdenvNoCC.mkDerivation rec {
```
Comment (drupol) [inline pkgs/applications/virtualization/tart/default.nix:5]: I think we should use the `finalAttrs` pattern here. Let me share a couple of useful links regarding this:

1. History: https://discourse.nixos.org/t/avoid-rec-expresions-in-nixpkgs/8293
2. Documentation: https://nixos.org/manual/nixpkgs/unstable/#mkderivation-recursive-attributes
3. Recent example of implementation: https://github.com/NixOS/nixpkgs/compare/17f96f7b978e61576cfe16136eb418f74fab9952..9e6ea843e473d34d4f379b8b0d8ef0425a06defe

**Comment by emilytrau** [comment]
> Thanks for reviving this PR. I found later on that I had an external dependency that was installed outside of nix that probably would need to be included too https://github.com/cirruslabs/softnet

Softnet looks like it requires a `setuid`-bit which usually makes things more complicated. On NixOS this is usually done with a module that creates a wrapper program (`security.wrappers.<name>.setuid`) but I'm not sure if there's a nix-darwin equivalent. The most generic way would probably to have the user install it outside nix as you had before.

**Comment by emilytrau** [comment]
I've added softnet as an optional dependency. It's from a release binary for now but I will keep an eye out for the Darwin SDK bump (https://github.com/NixOS/nixpkgs/pull/229210) and rewrite it as a source-build afterwards.

**Comment by emilytrau** [comment]
@khaled would you please give this a test and let me know if there's anything that doesn't meet your needs yet? :)

**Review by Enzime — APPROVED** [review]
LGTM, `tart` successfully creates and runs VMs on `aarch64-darwin`

**Review by drupol — APPROVED** [review]

**Pull request merged**